lib/neovim-plugin.nix

```diff
@@ -161,8 +162,41 @@
                   # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
 
-                  # Write the lua configuration `luaConfig.content` to the config file
-                  (setLuaConfig cfg.luaConfig.content)
+                  # Write the lua configuration `luaConfig.content` to the config file when lazy loading is not enabled
+                  (lib.mkIf (!cfg.lazyLoad.enable) (setLuaConfig cfg.luaConfig.content))
+
+                  # When lazy loading is enabled for this plugin, route its configuration to the enabled provider
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                      {
+                        assertion = cfg.lazyLoad.enable && (config.plugins.lz-n.enable || config.plugins.lazy.enable);
+                        message = "You have enabled lazy loading for ${originalName} but have not enabled any lazy loading plugins.";
+                      }
+                    ];
+                    plugins.lz-n = lib.mkIf config.plugins.lz-n.enable {
```

Comment:
Not sure I follow? Defining `lz-n.plugins` is effectivly a noop if `lz-n.enable = false`

Reply:
If i change the option type inside plugins.lz-n that is a breaking change, it doesn't matter if lz-n is enabled or not if we try to pass an incorrect type to an option.

Reply:
That's true, just felt like this was good defensive coding to avoid wasting time evaluating something that has no positive effect and only has possible negatives.

Reply:
> avoid wasting time evaluating something that has no positive effect and only has possible negatives

Worth noting that it will only be evaluated if `plugins.lz-n.plugins` itself gets evaluated, which I believe should only happen when `plugins.lz-n` is enabled? :upside_down_face:

Reply:
I created a test case to verify it will bomb out if i removed the condition and just passed configuration with an invalid option type. But, will use that as a smoke test if we do it in the future, for whatever reason.